src/components/Pama/pama.jsx

```diff
@@ -68,6 +68,27 @@ const dispatchRatingUpdates = (dispatch, updates) => updates
     rating,
   }));
 
+const dispatchResourceUpdate = (dispatch, update) => {
+  if (update.resource.code && update.resource.code.coding.length > 0) {
+    dispatch({
+      type: types.UPDATE_STUDY,
+      coding: update.resource.code.coding[0],
+    });
+  }
+  if (update.resource.reasonCode && update.resource.reasonCode.length > 0) {
+    dispatch({
```

Comment:
Might also want to dispatch an event that sets the rating, since the supplied suggestion may/should include rating extensions. (But this should automatically appear anyway after the subsequent service exchange round completes.)

Reply:
Consolidated into a new action in this scenario that sets code/reason/rating.